independent-projects/qute/core/src/main/java/io/quarkus/qute/EngineImpl.java

```diff
@@ -236,4 +238,30 @@ private Reader ensureBufferedReader(Reader reader) {
                         reader);
     }
 
+    @Override
+    public boolean hasTraceListeners() {
```

Comment:
I'm not sure how useful this method is, but this sounds like it would return `true` if anyone called `getTraceManager()` even without calling `addTraceListener()`, so it would lie a little bit.

Reply:
> At least forward the call `hasTraceListeners()` to the trace manager if it's not null?

Good catched! fixed